gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/ImmunizationMapperTest.java

```diff
@@ -183,6 +205,90 @@ public void mapObservationToImmunizationWithUNKVaccineCode() {
+            )
+        ).thenReturn(metaWithSecurity);
+        final var ehrExtract = unmarshallEhrExtract("immunization_with_ehrComposition_nopat_confidentiality_code.xml");
+
+        List<Immunization> immunizations = immunizationMapper.mapResources(
+            ehrExtract, getPatient(),
+            getEncounterList(), PRACTISE_CODE
+        );
+        var securityMeta = immunizations
+            .get(0)
+            .getMeta()
+            .getSecurity(NOPAT_URL_CODESYSTEM, NOPAT_CODE);
+
+        assertAll(
+            () -> {
+                assertThat(confidentialityCodeCaptor.getAllValues().get(0).isPresent())
+                    .isTrue();
+                assertThat(confidentialityCodeCaptor.getAllValues().get(0).get())
+                    .usingRecursiveComparison()
+                    .isEqualTo(getNoPatCV());
+            },
+            () -> {
+                assertThat(securityMeta).
+                    isNotNull();
+                assertThat(securityMeta.getDisplay())
+                    .isEqualTo(NOPAT_DISPLAY);
+            }
+        );
+    }
+
+    @Test
+    public void When_ObservationWithNoPatConfidentialityCode_Expect_MetaFromConfidentialityServiceWithSecurity() {
+        final var metaWithSecurity = MetaFactory.getMetaFor(META_WITH_SECURITY, META_PROFILE);
+        when(
+            confidentialityService.createMetaAndAddSecurityIfConfidentialityCodesPresent(
+                eq("Immunization-1"),
+                confidentialityCodeCaptor.capture(),
+                confidentialityCodeCaptor.capture()
+            )
+        ).thenReturn(metaWithSecurity);
+        final var ehrExtract = unmarshallEhrExtract("immunization_with_observation_nopat_confidentiality_code.xml");
+
+        List<Immunization> immunizations = immunizationMapper.mapResources(
+            ehrExtract, getPatient(),
+            getEncounterList(), PRACTISE_CODE
+        );
+        var securityMeta = immunizations
+            .get(0)
+            .getMeta()
+            .getSecurity(NOPAT_URL_CODESYSTEM, NOPAT_CODE);
+
+        assertAll(
+            () -> {
+                assertThat(confidentialityCodeCaptor.getAllValues().get(1).isPresent())
+                    .isTrue();
+                assertThat(confidentialityCodeCaptor.getAllValues().get(1).get())
+                    .usingRecursiveComparison()
+                    .isEqualTo(getNoPatCV());
+            },
+            () -> {
+                assertThat(securityMeta).
+                    isNotNull();
+                assertThat(securityMeta.getDisplay())
+                    .isEqualTo(NOPAT_DISPLAY);
+            }
+        );
+    }
+
+    private static @NotNull CV getNoPatCV() {
+        final var cv = new CV();
+        cv.setCode(ImmunizationMapperTest.NOPAT_CODE);
+        cv.setCodeSystem(ImmunizationMapperTest.NOPAT_OID_CODESYSTEM);
+        cv.setDisplayName(ImmunizationMapperTest.NOPAT_DISPLAY);
+        return cv;
+    }
```

Comment:
Hey, you can use some methods within `TestUtility` to scaffold CVs!

Reply:
Ahh coolio, didn't know the TestUtility has the create CV, was trying to find a method which already existed for it!